pisa/stages/xsec/dis_sys.py

```diff
@@ -0,0 +1,213 @@
+        bjorken_y
+            Inelasticity
+        dis
+            1 if event is DIS, else 0
+
+    """
+    def __init__(
+        self,
+        data=None,
+        params=None,
+        input_names=None,
+        output_names=None,
+        debug_mode=None,
+        input_specs=None,
+        calc_specs=None,
+        output_specs=None,
+        extrapolation_type='constant',
+    ):
+        expected_params = (
+            'dis_csms',
+        )
+        input_names = ()
+        output_names = ()
+
+        # what are the keys used from the inputs during apply
+        input_apply_keys = (
+            'dis_correction_total',
+            'dis_correction_diff',
+        )
+        # what keys are added or altered for the outputs during apply
+        output_apply_keys = (
+            'weights',
+        )
+
+        # init base class
+        super().__init__(
+            data=data,
+            params=params,
+            expected_params=expected_params,
+            input_names=input_names,
+            output_names=output_names,
+            debug_mode=debug_mode,
+            input_specs=input_specs,
+            calc_specs=calc_specs,
+            output_specs=output_specs,
+            input_apply_keys=input_apply_keys,
+            output_apply_keys=output_apply_keys,
+        )
+
+        assert self.input_mode is not None
+        assert self.calc_mode is None
+        assert self.output_mode is not None
+
+        self.extrapolation_type = extrapolation_type
+
+    @profile
+    def setup_function(self):
+
+        extrap_dict = from_file('cross_sections/tot_xsec_corr_Q2min1_isoscalar.pckl')
+
+        # load splines
+        wf_nucc =    from_file('cross_sections/dis_csms_splines_flat/NuMu_CC_flat.pckl')
+        wf_nubarcc = from_file('cross_sections/dis_csms_splines_flat/NuMu_Bar_CC_flat.pckl')
+        wf_nunc =    from_file('cross_sections/dis_csms_splines_flat/NuMu_NC_flat.pckl')
+        wf_nubarnc = from_file('cross_sections/dis_csms_splines_flat/NuMu_Bar_NC_flat.pckl')
+
+        # set this to events mode, as we need the per-event info to calculate these weights
+        self.data.data_specs = 'events'
+
+        for container in self.data:
+
+            # creat keys for external dict
+
+            if container.name.endswith('_cc'):
+                current = 'CC'
+            elif container.name.endswith('_nc'):
+                current = 'NC'
+            else:
+                raise ValueError('Can not determine whether container with name "%s" is pf type CC or NC based on its name'%container.name)
+            nu = 'Nu' if container['nubar'] > 0 else 'NuBar'
+
+            lgE = np.log10(container['true_energy'].get('host'))
+            bjorken_y = container['bjorken_y'].get('host')
+            dis = container['dis'].get('host')
+
+            #
+            # Calculate variation of total cross section
+            #
+
+            lgE_min = 2. if self.extrapolation_type == 'constant' else 1.68
```

Comment:
Unfortunately I can only find the number 1.68 in the file, but not 2......those must be some hand chosen, magic values...

Reply:
Right, I see 1.68 is in `tot_xsec_corr_Q2min1_isoscalar.pckl`, so that should be dynamically loaded. Can you find out from @jpyanez and/or Maria why the 2 magic number is chosen for constant extrapolation?

Reply:
Talking to juan pablo, this really is just a choice, and they tried different values between 50 and 100 GeV. So I now made this an argument to the function that defaults to 100